addons/pos_self_order/models/ir_http.py

```diff
@@ -33,16 +33,19 @@ def get_nearest_lang(self, lang_code: str) -> str:
             path_with_config = None
 
         if path_with_config:
-            config_id_match = re.search(r'/pos-self(?:/data)?/(\d+)', path_with_config)
-            if config_id_match:
+            config_id_match = re.search(r'/pos-self(?:/data)?/([a-zA-Z0-9]+)', path_with_config)
+            if config_id_match[1].isnumeric():
```

Comment:
Before when it was not matching anything, it was delegating to `super()`, this logic is gone in your work and I think it is a mistake
```suggestion
            if not config_id_match:
                return super().get_nearest_lang(lang_code)
            elif config_id_match[1].isnumeric():
```